--- a/lib/tasks/users.rake
+++ b/lib/tasks/users.rake
@@ -210,3 +210,12 @@ def find_user(username)
 
   user
 end
+
+desc "Anonymize user with the given username"
+task "users:anonymize", [:username] => [:environment] do |_, args|
+  username = args[:username]
+  user = find_user(username)
+  system_user = Discourse.system_user
+  UserAnonymizer.new(user, system_user).make_anonymous
+  puts "User #{username} anonymized"
+end
