--- a/lib/tasks/admin.rake
+++ b/lib/tasks/admin.rake
@@ -210,3 +210,12 @@ def find_user(username)
 
   user
 end
+
+desc "Promote a user to admin"
+task "admin:promote", [:email] => [:environment] do |_, args|
+email = args[:email]
+  if !email || email !~ /@/
+    puts "ERROR: Expecting rake admin:promote[some@email.com]"
+    exit 1
+  end
+  if User.find_by_email(email).nil?
+    puts "ERROR: User with email #{email} not found"
+    exit 1
+  end
+ admin = User.find_by_email(email)
+  grant_admin = ask("Do you want to grant Admin privileges to this account? (Y/n)  ")
+  if (grant_admin == "" || grant_admin.downcase == "y")
+    admin.grant_admin!
+    admin.change_trust_level!(1) if admin.trust_level < 1
+    admin.email_tokens.update_all confirmed: true
+    admin.activate
+
+    say("\nYour account now has Admin privileges!")
+  end
+end