--- a/lib/tasks/discourse-charm.rake
+++ b/lib/tasks/discourse-charm.rake
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+desc "Check if a user exists for given email address"
+task "users:exists", [:email] => [:environment] do |_, args|
+email = args[:email]
+  if !email || email !~ /@/
+    puts "ERROR: Expecting rake users:exists[some@email.com]"
+    exit 1
+  end
+  if User.find_by_email(email)
+    puts "ERROR: User with email #{email} not found"
+    exit 1
+  end
+  puts "User with email #{email} exists"
+end
