--- a/lib/tasks/discourse-charm.rake
+++ b/lib/tasks/discourse-charm.rake
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+desc "Check if a user exists for given email address"
+task "users:exists", [:email] => [:environment] do |_, args|
+  email = args[:email]
+  if User.find_by_email(email)
+    puts "User with email #{email} exists"
+    exit 0
+  end
+  puts "ERROR: User with email #{email} not found"
+  exit 2
+end
+
+desc "Activate a user account"
+task "users:activate", [:email] => [:environment] do |_, args|
+  email = args[:email]
+  user = User.find_by_email(email)
+  if !user
+    puts "User with email #{email} does not exist"
+    exit 2
+  end
+  user.email_tokens.update_all(confirmed: true)
+  puts "User with email #{email} activated"
+  exit 0
+end
