--- a/config/unicorn.conf.rb
+++ b/config/unicorn.conf.rb
@@ -52,6 +52,15 @@ check_client_connection false
 
 initialized = false
 before_fork do |server, worker|
+  Signal.trap 'TERM' do
+    puts 'Unicorn master intercepting TERM and sending myself QUIT after 5s instead'
+    Thread.new do
+      sleep 15
+      puts 'Send QUIT signal to master'
+      Process.kill 'QUIT', Process.pid
+    end
+  end
+
   unless initialized
     Discourse.preload_rails!
 
@@ -266,6 +275,9 @@ before_fork do |server, worker|
 end
 
 after_fork do |server, worker|
+  Signal.trap 'TERM' do
+    puts 'Unicorn worker intercepting TERM and doing nothing. Wait for master to sent QUIT'
+  end
   DiscourseEvent.trigger(:web_fork_started)
   Discourse.after_fork
 end
