diff --git a/config/pitchfork.conf.rb b/config/pitchfork.conf.rb
index 01e5f7fa..b3c20ed5 100644
--- a/config/pitchfork.conf.rb
+++ b/config/pitchfork.conf.rb
@@ -53,6 +53,15 @@ if ENV["RAILS_ENV"] != "production"
 end
 
 before_fork do |server|
+  Signal.trap "TERM" do
+    puts "Pitchfork master intercepting TERM and sending myself QUIT after 15s instead"
+    Thread.new do
+      sleep 15
+      puts "Send QUIT signal to master"
+      Process.kill "QUIT", Process.pid
+    end
+  end
+
   Discourse.redis.close
 
   throttle_time = Float(ENV["APP_SERVER_FORK_THROTTLE"], exception: false) || 1
@@ -85,6 +94,10 @@ end
 oob_gc_enabled = ENV["DISCOURSE_DISABLE_MAJOR_GC_DURING_REQUESTS"] && RUBY_VERSION >= "3.4"
 
 after_worker_fork do |server, worker|
+  Signal.trap "TERM" do
+    puts "Pitchfork worker intercepting TERM and doing nothing. Wait for master to send QUIT"
+  end
+
   DiscourseEvent.trigger(:web_fork_started)
   Discourse.apply_worker_db_variables_overrides
   Discourse.after_fork
