diff --git a/config/environments/production.rb b/config/environments/production.rb
index a523888a8d..7f41985b17 100644
--- a/config/environments/production.rb
+++ b/config/environments/production.rb
@@ -14,7 +14,7 @@ Discourse::Application.configure do
   # Disable Rails's static asset server (Apache or nginx will already do this)
   config.public_file_server.enabled = GlobalSetting.serve_static_assets || false
 
-  config.assets.js_compressor = :uglifier
+  config.assets.js_compressor = :terser
 
   # stuff should be pre-compiled
   config.assets.compile = false
